File: samples/SampleCounterV3/Mvu/Widgets/Column.cs
using Metano.TypeScript.DOM;

namespace SampleCounterV3.Mvu.Widgets;

public sealed class Column(int gap, Widget[] children) : Widget
{
    public override HtmlElement Render()
    {
        var div = Js.Document.CreateElement(HtmlElementType.Div);
        div.ApplyStyle($"display:flex;flex-direction:column;gap:{gap}px");

        foreach (var child in children)
        {
            div.Append(child.Render());
        }

        return div;
    }
}

File: targets/js/sample-counter-v3/src/mvu/widgets/column.ts
import type { IWidget } from "#/mvu";

export class Column implements IWidget {
  private readonly _children: IWidget[];

  constructor(children: IWidget[]) {
    this._children = children;
  }

  build(): HTMLElement {
    const div = document.createElement("div");
    for (let i = 0; i < this._children.length; i++) {
      div.append(this._children[i].build());
    }

    return div;
  }
}
